llama_stack/providers/inline/agents/meta_reference/agent_instance.py

```diff
@@ -614,117 +614,135 @@ async def _run(
                     logger.debug(f"completion message with EOM (iter: {n_iter}): {str(message)}")
                     input_messages = input_messages + [message]
             else:
-                logger.debug(f"completion message (iter: {n_iter}) from the model: {str(message)}")
-                # 1. Start the tool execution step and progress
-                step_id = str(uuid.uuid4())
-                yield AgentTurnResponseStreamChunk(
-                    event=AgentTurnResponseEvent(
-                        payload=AgentTurnResponseStepStartPayload(
-                            step_type=StepType.tool_execution.value,
-                            step_id=step_id,
+                input_messages = input_messages + [message]
```

Comment:
all of this code inline in this method is becoming quite hard to scan through. I think we need another file where we write some simple functions (all of these inside the `agent_instance` object is also a FAIL mode that I had long before initiated ever since our first agent impl lol)